core-bundle/src/Command/ResizeImagesCommand.php

```diff
@@ -0,0 +1,243 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Command;
+
+use Contao\CoreBundle\Image\ImageFactoryInterface;
+use Contao\Image\DeferredImageInterface;
+use Contao\Image\DeferredImageStorageInterface;
+use Contao\Image\DeferredResizerInterface;
+use Contao\Image\ResizerInterface;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\InvalidArgumentException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Terminal;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Process\PhpExecutableFinder;
+use Symfony\Component\Process\Process;
+
+/**
+ * Resize deferred images that have not been processed yet.
+ */
+class ResizeImagesCommand extends Command
+{
+    /**
+     * @var Filesystem
+     */
+    private $filesystem;
+
+    /**
+     * @var ImageFactoryInterface
+     */
+    private $imageFactory;
+
+    /**
+     * @var ?DeferredResizerInterface
+     */
+    private $resizer;
+
+    /**
+     * @var string
+     */
+    private $targetDir;
+
+    /**
+     * @var DeferredImageStorageInterface
+     */
+    private $storage;
+
+    /**
+     * @var int
+     */
+    private $terminalWidth;
+
+    public function __construct(Filesystem $filesystem, ImageFactoryInterface $imageFactory, ResizerInterface $resizer, string $targetDir, DeferredImageStorageInterface $storage)
```

Comment:
I guess you're injecting the `Filesystem` here to then be able to write unit tests. That's good but I'd make it the last and argument and optional in the constructor then.

Reply:
0bcdc86692017cf805102e36dc6122d984357550